src/lib.rs

```diff
@@ -63,31 +68,22 @@ pub fn current_millis() -> u64 {
     get_systimer_count() / (TICKS_PER_SECOND / 1000)
 }
 
-// TODO: should the below code live somewhere else, in its own module maybe? Or is it fine here?
-
-#[global_allocator]
-pub(crate) static ALLOCATOR: esp_alloc::EspHeap = esp_alloc::EspHeap::empty();
-
-pub fn init_heap() {
-    #[cfg(not(coex))]
-    const HEAP_SIZE: usize = 64 * 1024;
-
-    #[cfg(coex)]
-    const HEAP_SIZE: usize = 96 * 1024;
+#[cfg(not(coex))]
+const HEAP_SIZE: usize = 64 * 1024;
 
-    extern "C" {
-        static mut _heap_start: u32;
-        //static mut _heap_end: u32; // XXX we don't have it on ESP32-C3 currently
-    }
+#[cfg(coex)]
+const HEAP_SIZE: usize = 96 * 1024;
 
-    unsafe {
-        let heap_start = &_heap_start as *const _ as usize;
+static mut HEAP_DATA: [MaybeUninit<u8>; HEAP_SIZE] = [MaybeUninit::new(0u8); HEAP_SIZE];
```

Comment:
Using `uninit()` is still safe (and `const`!) but avoids unnecessary initialization.
```suggestion
static mut HEAP_DATA: [MaybeUninit<u8>; HEAP_SIZE] = [MaybeUninit::uninit(); HEAP_SIZE];
```

Reply:
Thanks! `MaybeUninit::new` is also const but comes with slightly higher costs. Actually, the array ends up in `.bss` so it's technically the same but using `uninit` expresses the intention better 👍